### DISCUSSION TITLE
beaker.hyperdrive.writeFile() not working in an <iframe>

### DISCUSSION THREAD

**Discussion by sentamalin** [beakerbrowser/beaker#1795] [General] [answered]
Good morning, everyone:

`beaker.hyperdrive.writeFile` causes an "Error: Uncaught Error" when it's called from inside of an `<iframe>`. At first, I figured it may be an error on my end since on the thing I've been working on the `<iframe>` was in a different drive and the parent window requested a write using `window.postMessage`; and I assumed that maybe I violated same-origin or some other security implementation. However, trying it in a test drive in same-origin causes the same error.

I realize after searching through closed issues and discussions that there have been conversations already regarding `<iframe>`; as such I've been hesitant to open an issue that's already known. Is this current expected behavior--and, as such, did I just blank out on something I needed to make `beaker` API calls like `hyperdrive` or `peersockets` work in `<iframe>` elements--or is this actual wonky behavior that shouldn't be happening?

Test Drive: hyper://d3b956ab90026da6975add81fcccab00f81590fee471c633e590abb7b3f3ca98/iframe/
Video: hyper://d3b956ab90026da6975add81fcccab00f81590fee471c633e590abb7b3f3ca98/writeFileError.webm

**Comment by pfrazee** [comment, marked as answer]
Sorry for the very long reply time, I've been in a heads-down sprint for over a month!

You shouldn't be getting "Uncaught Error" so I'll check into that, but it's currently correct that it fails: iframes are only allowed to access a subset of `hyperdrive` methods (reads). This is a temporary security measure while we figure out how to do permission modals for iframes

**Reply by sentamalin** [reply]
No worries; I get how busy things can get.

Is the current security measures also an explanation why drives in an `<iframe>` also can't utilize `beaker.peersockets` even if it works correctly in its own tab or that something going on in code in my end?

**Reply by pfrazee** [reply]
Yep that's what's going on there
